esp/GBPlay/main/hardware/wifi.c

```diff
@@ -13,16 +13,18 @@
 
 #define WIFI_SAVED_NETWORKS_STORAGE_KEY "wifi_networks"
 
-typedef enum {
-    CONNECTION_FAIL    = 1,
-    CONNECTION_SUCCESS = 2
-} WifiConnectionEvent;
+// Assumes dst is a statically allocated array
+#define TRUNCATED_STRING_COPY(dst, src) \
+    strncpy(dst, src, sizeof(dst)); \
+    dst[sizeof(dst) - 1] = '\0';
 
 typedef struct {
     wifi_network_credentials networks[WIFI_MAX_SAVED_NETWORKS];
     int count;
 } wifi_saved_network_info;
 
+ESP_EVENT_DEFINE_BASE(NETWORK_EVENT);
```

Comment:
Switched to using custom events here so we have more control, and to further abstract away the ESP Wi-Fi internals.